homeassistant/components/lastfm/sensor.py

```diff
@@ -35,23 +41,44 @@ def format_track(track: Track) -> str:
     return f"{track.artist} - {track.title}"
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the Last.fm sensor platform."""
-    lastfm_api = LastFMNetwork(api_key=config[CONF_API_KEY])
-    entities = []
-    for username in config[CONF_USERS]:
-        try:
-            user = lastfm_api.get_user(username)
-            entities.append(LastFmSensor(user, lastfm_api))
-        except WSError as exc:
-            LOGGER.error("Failed to load LastFM user `%s`: %r", username, exc)
-            return
-    add_entities(entities, True)
+    """Set up the Last.fm sensor platform from yaml."""
+
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml",
+        breaks_in_ha_version="2023.7.0",
```

Comment:
```suggestion
        breaks_in_ha_version="2023.8.0",
```

Reply:
Where can I find these guidelines?

Reply:
Not sure it's mentioned anywhere but for this it's always 2 releases so it can be in .6 and therefore it can only break earliest in .8